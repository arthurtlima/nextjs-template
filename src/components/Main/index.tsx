type MainProps = {
  title?: string
}

export default function Main({ title = 'test' }: MainProps) {
  return (
    <div>
      <h2>{title}</h2>
    </div>
  )
}
