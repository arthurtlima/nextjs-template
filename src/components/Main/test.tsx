import { screen, render } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render with default title', () => {
    const { container } = render(<Main />)
    expect(screen.getByRole('heading', { name: 'test' })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with custom title', () => {
    const { container } = render(<Main title="title" />)
    expect(screen.getByRole('heading', { name: 'title' })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
