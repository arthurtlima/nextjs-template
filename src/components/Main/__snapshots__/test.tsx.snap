// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render with custom title 1`] = `
<div>
  <h2>
    title
  </h2>
</div>
`;

exports[`<Main /> should render with default title 1`] = `
<div>
  <h2>
    test
  </h2>
</div>
`;
